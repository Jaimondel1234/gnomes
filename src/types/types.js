export const types = {
  login: "[Auth] Login",
  logout: "[Auth] Logout",
  loginRequest: "[Auth] Login Request",
};
