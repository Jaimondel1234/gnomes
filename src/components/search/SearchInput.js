import React from "react";
import {
  ContainerStyle,
  InputStyle,
  TextStyle,
  SpanStyle,
} from "../../styles/core";
import { useForm } from "../../hooks/useForm";
import styled from "styled-components";
import { useDispatch } from "react-redux";
import { startGnomeSearch } from "../../actions/gnome";

/** Style of input of search */
const SearchInputStyle = styled(InputStyle)`
  border-radius: 7px;
  font-size: 21px;
  font-weight: bold;
  margin-bottom: 20px;
  padding: 12px;
`;

export const SearchInput = () => {
  const dispatch = useDispatch();
  const [formValues, handleInputChange] = useForm({
    searchText: "",
  });

  const { searchText } = formValues;

  const handleSearch = (e) => {
    if (event.key === "Enter") {
      dispatch(startGnomeSearch(searchText));
    }
  };

  return (
    <SearchInputStyle
      type="text"
      placeholder="Tobus Quickwhistle"
      name="searchText"
      autoComplete="off"
      value={searchText}
      onChange={handleInputChange}
      onKeyPress={handleSearch}
    />
  );
};
