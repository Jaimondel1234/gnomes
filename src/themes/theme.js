/* 
  Theme
*/

export const theme = {
  color: {
    main: '#007bff',
  },
};
