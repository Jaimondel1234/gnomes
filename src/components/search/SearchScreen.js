import React from 'react';
import { ContainerStyle, TextStyle } from '../../styles/core';
import { SearchInput } from './SearchInput';
import { useSelector } from 'react-redux';
import { GnomeInformation } from '../Shared/GnomeInformation';
import { isEmpty } from 'lodash';
/*
component that displays the search screen for a gnome
*/
export const SearchScreen = () => {
  const { gnomeSearched } = useSelector((state) => state.gnome);
  return (
    <ContainerStyle m='20px'>
      <TextStyle fontSize='30px' fontWeight='bold' mb='10px'>
        Search the gnome:
      </TextStyle>
      <SearchInput />
      {!isEmpty(gnomeSearched) && <GnomeInformation {...gnomeSearched} />}
    </ContainerStyle>
  );
};
