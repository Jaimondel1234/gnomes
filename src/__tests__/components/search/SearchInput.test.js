import React from "react";
import { mount } from "enzyme";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import { ThemeProvider } from "styled-components";
import { theme } from "../../../themes/theme";
import { types } from "../../../types/types";

import { SearchInput } from "../../../components/search/SearchInput";

describe("Test <SearchInput/>", () => {
  const middlewares = [];
  const mockStore = configureStore(middlewares);
  const initState = {};

  const store = mockStore(initState);
  store.dispatch = jest.fn();
  const wrapper = mount(
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <SearchInput />
      </ThemeProvider>
    </Provider>
  );
  test("it must be displayed correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("if the enter key is pressed, it will trigger the gnome search action ", () => {
    wrapper.find("input").prop("onKeyPress")({ key: "Enter" });
    expect(store.dispatch).toHaveBeenCalledWith({
      type: types.startGnomeSearch,
      gnomeName: "",
    });
  });
});
