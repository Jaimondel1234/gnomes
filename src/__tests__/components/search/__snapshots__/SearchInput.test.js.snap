// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <SearchInput/> it must be displayed correctly 1`] = `
<input
  autoComplete="off"
  className="core__InputStyle-qu5lcc-5 SearchInput__SearchInputStyle-uz7aty-0 eMTzQF GMxWa"
  name="searchText"
  onChange={[Function]}
  onKeyPress={[Function]}
  placeholder="Tobus Quickwhistle"
  type="text"
  value=""
/>
`;
